bin/PerlDDS/Run_Test.pm

```diff
@@ -455,6 +455,132 @@ sub new {
+  }
+
+  # If /proc/sys/kernel/core_uses_pid is non-zero and the pattern
+  # doesn't have %p, then .PID is appended to the core file name.
+  my $uses_pid_file = "/proc/sys/kernel/core_uses_pid";
+  my $uses_pid = 0;
+  if (!open(my $uses_pid_fh, "<", "$uses_pid_file")) {
+    printf STDOUT "WARNING: Could not open $uses_pid_file\n";
+  } else {
+    $line = <$uses_pid_fh>;
+    if ($line ne "" || $line ne "\n") {
+      $uses_pid = $line;
+    }
+    close($uses_pid_fh);
+  }
+
+  my $exec_path = $process->Executable();
+
+  my $exec_name_idx = index($pattern, "%e");
+  if ($exec_name_idx != -1) {
+    my $exec_name = File::Basename::basename($exec_path);
+    # The core file name contains at most 15 characters from the executable name
+    # (https://man7.org/linux/man-pages/man5/core.5.html).
+    $exec_name = substr($exec_name, 0, 15);
+    substr($pattern, $exec_name_idx, 2) = $exec_name;
+  }
+
+  my $hname_idx = index($pattern, "%h");
+  if ($hname_idx != -1) {
+    substr($pattern, $hname_idx, 2) = Sys::Hostname::hostname();
+  }
+
+  my $pid_idx = index($pattern, "%p");
+  if ($pid_idx != -1) {
+    substr($pattern, $pid_idx, 2) = _getpid($process);
+  } elsif ($uses_pid != 0) {
+    $pattern = $pattern . "." . _getpid($process);
+  }
+
+  my $timestamp_idx = index($pattern, "%t");
+  my $core_file_path;
+  if ($timestamp_idx != -1) {
+    my $prefix = substr($pattern, 0, $timestamp_idx);
+    my $suffix_len = length($pattern) - $timestamp_idx - 2;
+    my $suffix = substr($pattern, $timestamp_idx + 2, $suffix_len);
+
+    # Get the core file with latest timestamp.
+    opendir(my $dh, $path);
+    my @files = grep(/$prefix[0-9]+$suffix/, readdir($dh));
+    my $latest_timestamp;
+    my $chosen_core_file;
+    foreach my $file (@files) {
+      my $timestamp_len = length($file) - $timestamp_idx - $suffix_len;
+      my $timestamp = substr($file, $timestamp_idx, $timestamp_len);
+      if (!defined $latest_timestamp) {
+        $latest_timestamp = $timestamp;
+        $chosen_core_file = $file;
+      } elsif ($latest_timestamp < $timestamp) {
+        $latest_timestamp = $timestamp;
+        $chosen_core_file = $file;
+      }
+    }
+    closedir($dh);
+    if (defined $chosen_core_file) {
+      $core_file_path = $path . "/" . $chosen_core_file;
+    } else {
+      printf STDOUT "WARNING: Could not determine a core file with timestamp\n";
+      return;
+    }
+  } else {
+    $core_file_path = $path . "/" . $pattern;
+  }
+
+  if (!(-e $core_file_path)) {
+    printf STDOUT "WARNING: Core file $core_file_path does not exist\n";
+    return;
+  }
+
+  # Print stack trace.
+  my $stack_trace;
+  if (system("gdb --version") != -1) {
+    $stack_trace = `gdb $exec_path -c $core_file_path -ex bt -ex quit`;
+  } elsif (system("lldb --version") != -1) {
+    printf STDOUT "WARNING: Failed printing stack trace with gdb. Trying lldb...\n";
+    $stack_trace = `lldb $exec_path -c $core_file_path -o bt -o quit`;
```

Comment:
This looks like it will always use gdb as long as it's there. I think this is okay as a default, but it's preferable to use binary tools from the same toolchain as the compiler. So Linux builds using clang we should probably use lldb, even though these systems are also going to have gdb. There are also cases where the names could be different, like I have `lldb-13` on my personal system, but not a regular `lldb`.

So what I'm thinking is this should probably be configurable, I guess as environment variables, so that we could force lldb on Linux builds using clang or set a specific lldb or gdb to use.

Reply:
That makes sense. I made changes to the PR to ACE. Users can now set the debugger they want with an environment variable.